test/notebooks/notebook_test_utils.py

```diff
@@ -0,0 +1,140 @@
+            hack_cell = nbformat.v4.new_code_cell(hack_content)
+            nb.cells.insert(cell_no + 1, hack_cell)
+
+
+def run_notebook(notebook_file: str, store_file: str, store_password: str,
+                 timeout: int = -1, hacks: Optional[List[Tuple[str, str]]] = None) -> None:
+    """
+    Executes notebook with added access to the configuration store.
+
+    Parameters:
+        notebook_file:  Notebook file.
+        store_file:     Configuration store file.
+        store_password: Configuration store password.
+        timeout:        Optional timeout in seconds
+        hacks:          Optional hacks to be inserted into the notebook
+                        before running it.
+    """
+
+    nb = nbformat.read(notebook_file, as_version=4)
+
+    if hacks:
+        _insert_hacks(nb, hacks)
+
+    # Insert the following code at the beginning of the notebook
+    init_code = f'''
+    def init_notebook_test():
+        from pathlib import Path
+        from exasol.secret_store import Secrets
+        global sb_config
+        sb_config = Secrets(Path("{store_file}"), "{store_password}")
+    init_notebook_test()
+    '''
+    nb.cells.insert(0, nbformat.v4.new_code_cell(init_code))
+
+    # Execute the notebook object, expecting to get no exceptions.
+    nb_client = NotebookClient(nb, timeout=timeout, kernel_name='python3')
+    nb_client.execute()
+
+
+@pytest.fixture
+def access_to_temp_secret_store(tmp_path: Path) -> Tuple[Path, str]:
+    """
+    Creates a temporary configuration store.
+    Brings up and subsequently destroys the Docker-DB.
+    Returns the temporary configuration store path and password.
+    """
+
+    # Create temporary secret store
+    store_path = tmp_path / 'tmp_config.sqlite'
+    # Should the password be constant or randomly generated is debatable.
+    # Strictly speaking, random password makes a test non-deterministic.
+    # On the other hand, it improves the security. If for some reason the
+    # temporary configuration store is not deleted, there will be no access
+    # to it since the password will be forgotten once the session is over.
+    store_password = generate_password(12)
+    secrets = Secrets(store_path, master_password=store_password)
+
+    # Set the configuration required by the ITDE manager and those the
+    # manager will not set after starting the Docker-DB.
+    _init_secret_store(secrets)
+
+    # Start the Docker-DB and then destroy it after the test finishes.
+    bring_itde_up(secrets)
+    try:
+        yield store_path, store_password
+    finally:
+        take_itde_down(secrets)
+
+
+@pytest.fixture
+def notebook_runner(access_to_temp_secret_store) -> Callable:
+    """
+    A fixture for running a notebook.
+    """
+
+    store_path, store_password = access_to_temp_secret_store
+    return partial(run_notebook, store_file=str(store_path), store_password=store_password)
+
+
+@pytest.fixture
+def uploading_hack() -> Tuple[str, str]:
```

Comment:
ok know I understand why you need to inject a hack

Reply:
maybe mention, that  extracting the model in bucketfs takes a while, however the upload returns immediately

Reply:
we also document that with the hacks in the developer guide